src/main/java/org/opensearch/flowframework/rest/AbstractSearchWorkflowAction.java

```diff
@@ -88,6 +92,20 @@ protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient cli
         searchSourceBuilder.fetchSource(getSourceContext(request, searchSourceBuilder));
         searchSourceBuilder.seqNoAndPrimaryTerm(true).version(true);
         searchSourceBuilder.timeout(flowFrameworkSettings.getRequestTimeout());
+
+        // Apply credential filter when searching templates
+        if (index.equals(GLOBAL_CONTEXT_INDEX)) {
+            searchSourceBuilder.scriptField(
+                "filter",
+                new Script(
+                    ScriptType.INLINE,
+                    "painless",
+                    "def filteredSource = new HashMap(params._source); def workflows = filteredSource.get(\"workflows\"); if (workflows != null) { def provision = workflows.get(\"provision\"); if (provision != null) { def nodes = provision.get(\"nodes\"); if (nodes != null) { for (node in nodes) { def userInputs = node.get(\"user_inputs\"); if (userInputs != null) { userInputs.remove(\"credential\"); } } } } } return filteredSource;",
```

Comment:
We should build a general JSON transformer so we don't have to do painless. ;)

(j/k, obvs.)